src/parser/ParsedQuery.cpp

```diff
@@ -272,16 +279,18 @@ bool ParsedQuery::GraphPattern::addLanguageFilter(const Variable& variable,
            stdv::filter(ad::toBool)) {
     for (auto& triple : basicPattern->_triples) {
       if (triple.o_ == variable &&
-          (triple.p_.operation_ == PropertyPath::Operation::IRI &&
-           !isVariable(triple.p_)) &&
-          !triple.p_.iri_.starts_with(
+          (std::holds_alternative<PropertyPath>(triple.p_) &&
+           std::get<PropertyPath>(triple.p_).operation_ ==
+               PropertyPath::Operation::IRI) &&
+          !std::get<PropertyPath>(triple.p_).iri_.starts_with(
```

Comment:
Think of something to make the code nicer... 
At least don't repeat the types, 
but I prefer some nicer functions that you directly implement in the `Triple` class
(maybe the predicate is not a plain variant, but something with nice functions).